### PR TITLE
Data Collection Practices

### PR DESCRIPTION
**Pull request by bpoplauschi**
### New Pull Request Checklist

* [X] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [X] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [X] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

This merge request fixes / refers to the following issues: #1187

### Pull Request Description

Add README section for the Data Collection Practices

### CODE REVIEW

**Comment by bpoplauschi** [comment]
@CocoaLumberjack/collaborators let's see some opinions here on how we can add more value to our framework's users with this section of the docs.

File: README.md [hunk 1]
```diff
@@ -213,7 +213,45 @@ The current version of Lumberjack requires:
 - If you **have a feature request**, open an issue.
 - If you **want to contribute**, submit a pull request.
 
+### Data Collection Practices
+
+Per  [App privacy details on the App Store](https://developer.apple.com/app-store/app-privacy-details/), Apple is requesting app developers to provide info about their data collection, us SDK maintainers must provide them with the same data.
+
+#### Data collection by the framework
+
+**By default, CocoaLumberjack does not collect any data on its own.**
```
Comment (lolgear) [inline README.md:222]: Maybe it is better to emphasize this via quote? If you open "Display rich diff" view, you will find that two bold lines one below another will interfere and will hide second line.

File: README.md [hunk 2]
```diff
@@ -213,7 +213,45 @@ The current version of Lumberjack requires:
...
+- sensitive info
+- user content
+- history (browsing, search, ...)
+- purchases
+- diagnostics
+- ...
+
+Example: `DDLogInfo("User: \(myUser)")` will add the `myUser` info to the logs, so if those are forwarded to a 3rd party or sent via email, that may qualify as data collection.
+
+**The important thing is you app developers are fully responsible for adding sensitive data to the logs, thus documenting that via the Data Collection Practices info!**
```
Comment (lolgear) [inline README.md:248]: Here, I would like to add section Important?

### Important note

```
App developers are fully responsible...
```

File: README.md (hunk 1 above)
Comment (lolgear) [reply README.md:222]: Something like this

#### Data collection by the framework

```
 By default, CocoaLumberjack does NOT collect any data on its own.
```

Comment (lolgear) [reply README.md:222]: Still not the best option, but, it separates header and body.

Comment (bpoplauschi) [reply README.md]: I have converted all our headings from h3 to h2 (which was available), thus making "Data collection by the framework" an h3 which is much more distinct than an h4. Let me know if this is ok now

File: README.md (hunk 2 above)
Comment (bpoplauschi) [reply README.md]: Good idea, I have changed that am moved it up

Comment (lolgear) [reply README.md]: I still not see it. Look at `rich diff view`.

If I read it in text, I won't spend enough attention/time on it. But I should.
**Important note: app developers are fully responsible for any sensitive data collected through our logging system!**

**Review by sushichop — COMMENTED** [review]
Great work! IMO, this is enough👍

**Review by sushichop — APPROVED** [review]

**Comment by bpoplauschi** [comment]
Cool @sushichop, thanks for looking into this.
I would like to also get @ffried's opinion before merging, just to be on the safe side :)

**Comment by sushichop** [comment]
@bpoplauschi 

> I would like to also get @ffried's opinion before merging, just to be on the safe side :)

I feel the same way🙂

**Comment by bpoplauschi** [comment]
@sushichop I just wanted both your opinions before merging ;)

**Comment by sushichop** [comment]
@bpoplauschi 

> @sushichop I just wanted both your opinions before merging ;)

Yeah, Thanks. Let's wait for a response🙂

File: README.md [hunk 3]
```diff
@@ -205,19 +205,61 @@ The current version of Lumberjack requires:
...
 - If you **need help**, use [Stack Overflow](http://stackoverflow.com/questions/tagged/lumberjack). (Tag 'lumberjack')
 - If you'd like to **ask a general question**, use [Stack Overflow](http://stackoverflow.com/questions/tagged/lumberjack).
 - If you **found a bug**, open an issue.
 - If you **have a feature request**, open an issue.
 - If you **want to contribute**, submit a pull request.
 
-### Author
+## Data Collection Practices
+
+Per  [App privacy details on the App Store](https://developer.apple.com/app-store/app-privacy-details/), Apple is requesting app developers to provide info about their data collection, us SDK maintainers must provide them with the same data.
```
Comment (ffried) [inline README.md:218]:
```suggestion
Per [App privacy details on the App Store](https://developer.apple.com/app-store/app-privacy-details/), Apple is requesting app developers to provide info about their data collection, us SDK maintainers must provide them with the same data.
```

File: README.md [hunk 4]
```diff
@@ -205,19 +205,61 @@ The current version of Lumberjack requires:
...
+
+**By default, CocoaLumberjack does NOT collect any data on its own.** 
+
+[See our Data Collection Practices list.](https://cocoalumberjack.github.io/DataCollection/index.html)
+
+### Indirect data collection through the framework
+
+CocoaLumberjack is a logging framework which makes it easy to send those logs to different platforms.
+
+This is why collecting  data might happen quite easily, if the app developers include any sensitive data into the logs.
```
Comment (ffried) [inline README.md:230]:
```suggestion
This is why collecting data might happen quite easily, if app developers include any sensitive data in their log messages.
```

**Review by ffried — COMMENTED** [review]
Looks good in general. 👍 
Just found two superfluous whitespaces and suggested a small re-phrasing of one sentence.

**Comment by lolgear** [comment]
@bpoplauschi 
Am I right that you said about "docs" in related issue, but this PR only affects Readme.
Not sure if Readme is a document that is convenient for app developers.
Should we add additional document about data collection?

File: README.md (hunk 3 above)
Comment (bpoplauschi) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (bpoplauschi) [reply README.md]: Done

**Comment by bpoplauschi** [comment]
@ffried thanks for taking a look, I agree with your suggestions and have fixed those in a commit.
@lolgear once this PR is merged, I will regenerate the doc on https://cocoalumberjack.github.io which includes the content from our Readme, so that will also live on our documentation main page. I think this is enough.

**Review by ffried — APPROVED** [review]

**Pull request merged**